### ISSUE TITLE
500 Server Error: /api/v1/uniprot/summary/A0A6P1NCC6

### ISSUE THREAD

**Issue by sillitoe** [UCLOrengoGroup/ted-web#55]
This errors with 500 (probably due to "na" being parsed as `float`)

https://ted.cathdb.info/api/v1/uniprot/summary/A0A6P1NCC6

```
GET /api/v1/uniprot/summary/A0A6P1NCC6 HTTP/1.1 500 Internal Server Error
```

```
ValueError: Out of range float values are not JSON compliant
```

```
backend-1  | INFO:     172.28.0.5:48812 - "GET /api/v1/uniprot/summary/A0A6P1NCC6 HTTP/1.1" 500 Internal Server Error
backend-1  | ERROR:    Exception in ASGI application
backend-1  | Traceback (most recent call last):
backend-1  |   File "/usr/local/lib/python3.10/site-packages/uvicorn/protocols/http/httptools_impl.py", line 426, in run_asgi
backend-1  |     result = await app(  # type: ignore[func-returns-value]
backend-1  |   File "/usr/local/lib/python3.10/site-packages/uvicorn/middleware/proxy_headers.py", line 84, in __call__
backend-1  |     return await self.app(scope, receive, send)
backend-1  |   File "/usr/local/lib/python3.10/site-packages/fastapi/applications.py", line 1054, in __call__
backend-1  |     await super().__call__(scope, receive, send)
backend-1  |   File "/usr/local/lib/python3.10/site-packages/starlette/applications.py", line 123, in __call__
backend-1  |     await self.middleware_stack(scope, receive, send)
backend-1  |   File "/usr/local/lib/python3.10/site-packages/starlette/middleware/errors.py", line 186, in __call__
backend-1  |     raise exc
backend-1  |   File "/usr/local/lib/python3.10/site-packages/starlette/middleware/errors.py", line 164, in __call__
backend-1  |     await self.app(scope, receive, _send)
backend-1  |   File "/usr/local/lib/python3.10/site-packages/starlette/middleware/cors.py", line 83, in __call__
backend-1  |     await self.app(scope, receive, send)
backend-1  |   File "/usr/local/lib/python3.10/site-packages/starlette/middleware/exceptions.py", line 62, in __call__
backend-1  |     await wrap_app_handling_exceptions(self.app, conn)(scope, receive, send)
backend-1  |   File "/usr/local/lib/python3.10/site-packages/starlette/_exception_handler.py", line 64, in wrapped_app
backend-1  |     raise exc
backend-1  |   File "/usr/local/lib/python3.10/site-packages/starlette/_exception_handler.py", line 53, in wrapped_app
backend-1  |     await app(scope, receive, sender)
backend-1  |   File "/usr/local/lib/python3.10/site-packages/starlette/routing.py", line 758, in __call__
backend-1  |     await self.middleware_stack(scope, receive, send)
backend-1  |   File "/usr/local/lib/python3.10/site-packages/starlette/routing.py", line 778, in app
backend-1  |     await route.handle(scope, receive, send)
backend-1  |   File "/usr/local/lib/python3.10/site-packages/starlette/routing.py", line 299, in handle
backend-1  |     await self.app(scope, receive, send)
backend-1  |   File "/usr/local/lib/python3.10/site-packages/starlette/routing.py", line 79, in app
backend-1  |     await wrap_app_handling_exceptions(app, request)(scope, receive, send)
backend-1  |   File "/usr/local/lib/python3.10/site-packages/starlette/_exception_handler.py", line 64, in wrapped_app
backend-1  |     raise exc
backend-1  |   File "/usr/local/lib/python3.10/site-packages/starlette/_exception_handler.py", line 53, in wrapped_app
backend-1  |     await app(scope, receive, sender)
backend-1  |   File "/usr/local/lib/python3.10/site-packages/starlette/routing.py", line 74, in app
backend-1  |     response = await func(request)
backend-1  |   File "/usr/local/lib/python3.10/site-packages/fastapi/routing.py", line 326, in app
backend-1  |     response = actual_response_class(content, **response_args)
backend-1  |   File "/usr/local/lib/python3.10/site-packages/starlette/responses.py", line 183, in __init__
backend-1  |     super().__init__(content, status_code, headers, media_type, background)
backend-1  |   File "/usr/local/lib/python3.10/site-packages/starlette/responses.py", line 41, in __init__
backend-1  |     self.body = self.render(content)
backend-1  |   File "/usr/local/lib/python3.10/site-packages/starlette/responses.py", line 186, in render
backend-1  |     return json.dumps(
backend-1  |   File "/usr/local/lib/python3.10/json/__init__.py", line 238, in dumps
backend-1  |     **kw).encode(obj)
backend-1  |   File "/usr/local/lib/python3.10/json/encoder.py", line 199, in encode
backend-1  |     chunks = self.iterencode(o, _one_shot=True)
backend-1  |   File "/usr/local/lib/python3.10/json/encoder.py", line 257, in iterencode
backend-1  |     return _iterencode(o, 0)
backend-1  | ValueError: Out of range float values are not JSON compliant
```

**Comment by sillitoe** [comment]
Backend fixed in #57, frontend still needs to be fixed:

<img width="593" alt="image" src="https://github.com/user-attachments/assets/8f165b32-4ac1-4792-af19-fce8a1a9b1f8">

**Comment by sillitoe** [comment]
fixed in #58
